paramiko/transport.py

```diff
@@ -3050,7 +3050,12 @@ def _set(self, name, orig, x):
         possible = list(getattr(self._transport, orig).keys())
         forbidden = [n for n in x if n not in possible]
         if len(forbidden) > 0:
-            raise ValueError("unknown cipher")
+            name_parsed = name.replace("_preferred_", "")
+            raise ValueError(
+                "unknown {} name: {}, possible options: {}".format(
+                    name_parsed, x, possible
```

Comment:
```pycon
>>> try:
...     opts.ciphers = ['unknown', 'aes128-ctr']
... except ValueError as e:
...     print(e)
...
unknown ciphers name: ('unknown', 'aes128-ctr'), possible options: ['aes128-ctr', 'aes192-ctr', 'aes256-ctr', 'aes128-cbc', 'aes192-cbc', 'aes256-cbc', '3des-cbc']
```

I think only "unknown" algorithms should be warned.


```diff
-                   name_parsed, x, possible
+                   name_parsed, forbidden, possible
```

Reply:
makes sense, addressed